gcm/client.go

```diff
@@ -0,0 +1,103 @@
+package gcm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// gcm Client const variables
+const (
+	DefaultGCMEndpoint = "https://fcm.googleapis.com/fcm/send"
+)
+
+// Client is GCM client
+type Client struct {
+	endpoint *url.URL
+	apiKey   string
+	Client   *http.Client
+}
+
+// Send sends notifications to gcm (TODO: send retry)
+func (gc *Client) Send(p Payload) (*Response, error) {
+	req, err := gc.NewRequest(p)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := gc.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	resp := &Response{}
+	dec := json.NewDecoder(res.Body)
+	if err = dec.Decode(&resp.Body); err != nil {
+		return nil, err
+	}
+	resp.Header.StatusCode = res.StatusCode
+
+	if res.StatusCode != http.StatusOK {
```

Comment:
I think, it is better to return immediately if this condition is true, because indentation will be less.